src/render/Renderer.cpp

```diff
@@ -2039,6 +2039,14 @@ bool CHyprRenderer::applyMonitorRule(CMonitor* pMonitor, SMonitorRule* pMonitorR
 
     wlr_output_set_scale(pMonitor->output, pMonitor->scale);
 
+    // Set scale for all surfaces on this monitor, needed for some clients
+    // but not on unsafe state to avoid crashes
+    if (!g_pCompositor->m_bUnsafeState) {
```

Comment:
last nit: I'd move this below. Probably after `wlr_damage_ring_set_bounds`, to avoid frankenstein states should anything change around here or in the func.